lib/SymbolGraphGen/SymbolGraphASTWalker.cpp

```diff
@@ -22,9 +22,11 @@ using namespace swift;
 using namespace symbolgraphgen;
 
 SymbolGraphASTWalker::SymbolGraphASTWalker(ModuleDecl &M,
+                                           const SmallPtrSet<ModuleDecl *, 4> ExportedImportedModules,
```

Comment:
If you use the sorted vector approach described in the comment on `collectParsedExportedImports` you won't have to hardcode the small size template parameter here. You can then pass the data via `SmallVectorImpl` or an `ArrayRef`. You would need to change the definition of `isFromExportedImportedModule` to use `std::lower_bound` to perform the binary search.